wc-vt-giftcards.php

```diff
@@ -425,29 +440,57 @@ public function init_form_fields() {
         );
       }
 
-      return array_slice($rows, 0, count($rows) - 1, true)
-        + $coupon_entries
-        + array_slice($rows, count($rows) - 1, 1, true);
+      return array_slice($rows, 0, count($rows) - 1, true) 
+	    + $coupon_entries 
+		+ array_slice($rows, count($rows) - 1, 1, true);
     }, 10, 2);
 
     // Woo integration - gift card form
     // --------------------------------
-
+	
     $cart_hook = apply_filters('woocommerce_valutec_gift_cards_hook', 'woocommerce_after_cart_totals');
+    $checkout_hook = apply_filters('woocommerce_valutec_gift_cards_hook', 'woocommerce_before_checkout_form');
 
     add_action($cart_hook, function () { // TODO what if plugin is deactivated during order
       if (WC()->cart->needs_payment()) {
         $button_classes = apply_filters('woocommerce_valutec_gift_cards_btnclass', ['apply-gift-card', 'button']);
         do_action('woocommerce_valutec_gift_cards_before_form');
-        ?><form action="<?php echo esc_url( WC()->cart->get_cart_url() ); ?>" method="post" class="apply-gift-card">
-          <input type="text" name="gift_card_code" value="" placeholder="Gift card number"/>
-          <input type="submit" class="<?php echo join(' ', $button_classes) ?>" name="update_cart" value="Apply gift card" />
-          <?php wp_nonce_field( 'woocommerce-cart' ); ?>
-        </form><?php
+        ?><div class="gift-cards" id="gift-cards">
+	      <p><strong>Have a gift card you'd like to use towards your purchase?</strong><br class="break">
+          Enter your gift card number below to apply towards your total purchase amount.</p>
+          <form action="<?php echo esc_url(WC()->cart->get_checkout_url()); ?>" method="post" class="apply-gift-card">
+          <input type="text" name="gift_card_code" value="" placeholder="Gift card number" />
+          <input type="text" name="gift_card_pin" value="" size="8" maxlength="8" placeholder="8 Digit PIN" />
+          <input type="submit" class="apply-gift-card button" name="update_cart" value="Apply Gift Card" />
+          <?php wp_nonce_field('woocommerce-cart'); ?>
+        </form></div><?php
         do_action('woocommerce_valutec_gift_cards_after_form');
       }
     });
 
+    add_action($checkout_hook, function () { // TODO what if plugin is deactivated during order
+      if (WC()->cart->needs_payment()) {
+        $button_classes = apply_filters('woocommerce_valutec_gift_cards_btnclass', ['apply-gift-card', 'button']);
+        do_action('woocommerce_valutec_gift_cards_before_form');
+        ?><div class="gift-cards" id="gift-cards">
+	      <p><strong>Have a gift card you'd like to use towards your purchase?</strong><br class="break">
+          Enter your gift card number below to apply towards your total purchase amount.</p>
```

Comment:
Does this add the gift card to another page? If so, fine with me but should we put the text above through a `do_filter` so it's customizable?